web/src/compose_actions.js

```diff
@@ -70,6 +71,38 @@ function show_compose_box(msg_type, opts) {
     // When changing this, edit the 42px in _maybe_autoscroll
     $(".new_message_textarea").css("min-height", "3em");
     compose_ui.set_focus(msg_type, opts);
+    const filter = narrow_state.filter();
+    if (filter && !filter.is_common_narrow()) {
+        show_compose_in_search_view_banner();
+    }
+}
+
+function show_compose_in_search_view_banner() {
+    // TODO: use compose_state.stream_id() when #25785 merges.
+    let stream_id = "";
+    const stream_name = compose_state.stream_name();
+    if (stream_name) {
+        const sub = stream_data.get_sub(stream_name);
+        if (sub) {
+            stream_id = sub.stream_id;
+        }
+
+        const search_view_banner = render_compose_banner({
+            banner_type: compose_banner.WARNING,
+            classname: compose_banner.CLASSNAMES.compose_in_search_view,
+            stream_id,
+            topic_name: compose_state.topic(),
```

Comment:
There is no need to pass around `stream_id` and `topic_name` in this case since these value case be changed by the user
and then "Go to conversation" will take user to the wrong narrow. You can use `compose_state` to get stream / topic / DM name when user clicks on "Go to conversation".

Reply:
That makes sense. However, my original commit didn't pass `stream_id` and `topic_name`. That change was inputted by @evykassirer, so I'm unsure whether I should amend it.

Reply:
Yeah sorry about that!